src/libraries/System.Text.Json/tests/Serialization/Null.ReadTests.cs

```diff
@@ -152,5 +156,37 @@ public static void NullAcceptsLeadingAndTrailingTrivia()
                 Assert.Null(obj);
             }
         }
+
+        [Fact]
+        public static void NullReadTestChar()
+        {
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("null"));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>(""));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("1234"));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("\"stringTooLong\""));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("\"\u0059\"B"));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<char>("\"\uD800\uDC00\""));
+            Assert.Equal('a', JsonSerializer.Deserialize<char>("\"a\""));
+            Assert.Equal('Y', JsonSerializer.Deserialize<char>("\"\u0059\""));
+        }
+
+        [ActiveIssue(1037)]
+        [Fact]
+        public static void ParseNullStringToStructShouldThrowJsonException()
+        {
+            string nullString = "null";
+            Utf8JsonReader reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(nullString));
+
+            JsonTestHelper.AssertThrows<JsonException>(reader, (reader) => JsonSerializer.Deserialize<SimpleStruct>(ref reader));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<SimpleStruct>(Encoding.UTF8.GetBytes(nullString)));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<SimpleStruct>(nullString));
+        }
+
+        [ActiveIssue(1037)]
+        [Fact]
+        public static async Task AsyncParseNullStringShouldThrowJsonException()
```

Comment:
nit: Rename to put `Async` at the end

```suggestion
        public static async Task ParseNullStringShouldThrowJsonExceptionAsync()
```